### PR TITLE
fix: Plain text paste does not preserve line breaks

### PR DESCRIPTION
**Pull request by Gxrvish**
# Summary

This PR aims to solve an issue described [here](https://github.com/TypeCellOS/BlockNote/issues/2693)

## Checklist

- [ ] Code follows the project's coding standards.
- [ ] Unit tests covering the new feature have been added.
- [ ] All existing tests pass.
- [ ] The documentation has been updated to reflect the new feature

### CODE REVIEW

**Comment by Gxrvish** [comment]
@YousefED few remarks?
first time solving bug in BlockNote, Sorry if i missed some internal guidelines :slightly_smiling_face: 
Thanks!

File: package.json [hunk 1]
```diff
@@ -85,5 +85,8 @@
     "docs",
     "shared",
     "tests"
-  ]
+  ],
+  "dependencies": {
+    "remark-breaks": "^4.0.0"
+  }
```
Comment (nperez0111) [inline package.json:91]: This is installed to the wrong package.json, it should be installed to the one in packages/core/package.json, this is a monorepo

File: packages/core/src/api/parsers/markdown/parseMarkdown.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import { Schema } from "prosemirror-model";
 import remarkGfm from "remark-gfm";
 import remarkParse from "remark-parse";
+import remarkBreaks from "remark-breaks";
```
Comment (nperez0111) [inline packages/core/src/api/parsers/markdown/parseMarkdown.ts:4]: Honestly, would prefer not to introduce a whole package just for this. And would like to have tests added that prove it does not break existing behavior

**Review by nperez0111 — CHANGES_REQUESTED** [review]

File: packages/core/src/api/parsers/markdown/parseMarkdown.ts (hunk 2 above)
Comment (Gxrvish) [reply packages/core/src/api/parsers/markdown/parseMarkdown.ts:4]: > Honestly, would prefer not to introduce a whole package just for this

For this operation adding a package is worth, Because doing it from scratch without the help of package might break some edge cases which may introduce unusual behaviour.

> would like to have tests added that prove it does not break existing behavior

Will add it soon :slightly_smiling_face: 
Thanks for the review :blush: !!!

**Comment by nperez0111** [comment]
Thanks for your efforts we've shipped a custom markdown parser which fixes this: https://github.com/TypeCellOS/BlockNote/pull/2624

**Comment by gppa-odoo** [comment]
@nperez0111 Thanks!
Will continue to ship good PRs, Had great learning.
